### PR TITLE
Updating glossary.md adding frequently used terms

### PR DESCRIPTION
**Pull request by Frostbite22**
Defining frequently used terms

#### What does this PR do?
Adding to the glossary the most frequently used terms that all contributors should be familiar with.

### CODE REVIEW

File: tips/glossary.md [hunk 1]
```diff
@@ -14,6 +14,29 @@ window.addEventListener("click", function () {
...
 ```
 
+### Frequently Used Terms Definitions 
+**git**: an open source, distributed version-control system
+
+**GitHub**: a platform for hosting and collaborating on Git repositories
+
+**commit**: a Git object, a snapshot of your entire repository compressed into a SHA
+
+**branch**: a lightweight movable pointer to a commit
```
Comment (schmelto) [inline tips/glossary.md:24]: I would not describe a branch like this cause a branch is a snapshot of the code in which multiple commits can be added etc.

File: tips/glossary.md [hunk 2]
```diff
@@ -14,6 +14,29 @@ window.addEventListener("click", function () {
...
+tests, and more
+
+**HEAD**: representing your current working directory, the HEAD pointer can be moved to different branches, tags, or commits
+when using git checkout
+
 ### Additional Resources
 
 [Anonymous Function](https://en.wikipedia.org/wiki/Anonymous_function)
+
+[Frequently Used Terms Definitions](services@github.com)
```
Comment (schmelto) [inline tips/glossary.md:42]: Why an email address?

Comment (Frostbite22) [reply tips/glossary.md:42]: that's the source of the defined terms, maybe if someone wants to learn more about Github or have any other request or issue could email them. I can discard it if you think it's unnecessary

File: tips/glossary.md (hunk 1 above)
Comment (Frostbite22) [reply tips/glossary.md:24]: branch :  is a snapshot of the code in which multiple commits can be added and get merged to the main branch 

what do you think ?

Comment (schmelto) [reply tips/glossary.md:24]: ok git documentation also refers to it as pointer to a commit :D (https://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell)

File: tips/glossary.md (hunk 2 above)
Comment (schmelto) [reply tips/glossary.md:42]: I would add a website or something like this.

File: tips/glossary.md (hunk 1 above)
Comment (Frostbite22) [reply tips/glossary.md:24]: technically, they're not wrong ; practically, we can explain it in more easily xD

File: tips/glossary.md (hunk 2 above)
Comment (Frostbite22) [reply tips/glossary.md:42]: https://docs.github.com/en/get-started/quickstart/github-glossary maybe it could help better

Comment (schmelto) [reply tips/glossary.md:42]: yes like that one (y)

File: tips/glossary.md (hunk 1 above)
Comment (schmelto) [reply tips/glossary.md:24]: you can choose what you want to write :)

**Review by schmelto — APPROVED** [review]

**Comment by schmelto** [comment]
**Thanks for your contribution** 🚀

**Pull request merged**